File: shared/models/media.py
"""
SQLAlchemy 模型定义 - Media
由 routes.yaml 自动生成 - 请勿手动修改
生成时间：2026-04-26 19:54:29
"""


from sqlalchemy import (BigInteger, Boolean, Column, DateTime, ForeignKey,
                        Integer, String, Text)

from . import Base  # 使用统一的 Base


class Media(Base):
    """媒体文件模型模型"""
    __tablename__ = 'media'


    id = Column(BigInteger, primary_key=True, autoincrement=True, doc='id')


    user = Column(BigInteger, ForeignKey('users.id'), nullable=False, doc='user')

    hash = Column(String(64), nullable=True, doc='hash')

    filename = Column(String(255), nullable=True, doc='filename')

    original_filename = Column(String(255), nullable=True, doc='original_filename')

    file_path = Column(String(500), nullable=True, doc='file_path')

    file_url = Column(String(500), nullable=True, doc='file_url')

    file_size = Column(BigInteger, default=0, doc='file_size')

    file_type = Column(String(255), default='other', doc='file_type')

    mime_type = Column(String(255), nullable=True, doc='mime_type')

    width = Column(BigInteger, nullable=True, doc='width')

    height = Column(BigInteger, nullable=True, doc='height')

    duration = Column(BigInteger, nullable=True, doc='duration')

    thumbnail_path = Column(String(255), nullable=True, doc='thumbnail_path')

    thumbnail_url = Column(String(255), nullable=True, doc='thumbnail_url')

    description = Column(String(255), nullable=True, doc='description')

    alt_text = Column(String(255), nullable=True, doc='alt_text')

    is_public = Column(Boolean, default=True, doc='is_public')

    download_count = Column(BigInteger, default=0, doc='download_count')

    category = Column(String(100), nullable=True, doc='category')

    tags = Column(String(500), nullable=True, doc='tags')

    folder_id = Column(BigInteger, ForeignKey('media_folders.id'), nullable=True, doc='folder_id')

    created_at = Column(DateTime, doc='created_at')

    updated_at = Column(DateTime, doc='updated_at')




    def to_dict(self, exclude_sensitive=True):
        """转换为字典
        
        Args:
            exclude_sensitive: 是否排除敏感字段（密码、密钥、token 等）
        """
        data = {
            'id': self.id,
            'user': self.user,
            'hash': self.hash,
            'filename': self.filename,
            'original_filename': self.original_filename,
            'file_path': self.file_path,
            'file_url': self.file_url,
            'file_size': self.file_size,
            'file_type': self.file_type,
            'mime_type': self.mime_type,
            'width': self.width,
            'height': self.height,
            'duration': self.duration,
            'thumbnail_path': self.thumbnail_path,
            'thumbnail_url': self.thumbnail_url,
            'description': self.description,
            'alt_text': self.alt_text,
            'is_public': self.is_public,
            'download_count': self.download_count,
            'category': self.category,
            'tags': self.tags,
            'folder_id': self.folder_id,
            'created_at': self.created_at.isoformat() if self.created_at else None,
            'updated_at': self.updated_at.isoformat() if self.updated_at else None,
        }

        # 只有当明确要求包含敏感字段时才添加
        if not exclude_sensitive:
            sensitive_data = {
            }
            data.update(sensitive_data)

        return data

    def __repr__(self):
        """字符串表示"""
        return f'<Media id={self.id}>'

File: frontend/web/src/api/modules/media.ts
/** 媒体库接口（/api/v3/content/media） */

import http from '../request'
import type {PageQuery, PageResult} from '../types'

export interface MediaItem {
  id: number
  user_id?: number | null
  filename?: string | null
  original_filename?: string | null
  file_url?: string | null
  file_size?: number | null
  mime_type?: string | null
  file_type?: string | null
  width?: number | null
  height?: number | null
  duration?: number | null
  thumbnail_url?: string | null
  description?: string | null
  alt_text?: string | null
  is_public: boolean
  download_count: number
  category?: string | null
  tags: string[]
  folder_id?: number | null
  created_at?: string | null
  updated_at?: string | null
}

export interface MediaFolder {
  id: number
  name: string
  parent_id?: number | null
  user_id?: number | null
  description?: string | null
  sort_order: number
  is_public: boolean
  media_count: number
  children: MediaFolder[]
}

export interface MediaQuery extends PageQuery {
  folder_id?: number
  mime_type?: string
  user_id?: number
  is_public?: boolean
  category?: string
}

export const mediaApi = {
  list: (params?: MediaQuery): Promise<PageResult<MediaItem>> =>
    http.page<MediaItem>('/content/media', params),
  detail: (id: number) => http.get<MediaItem>(`/content/media/${id}`),
  update: (
    id: number,
    data: Partial<{
      description: string
      alt_text: string
      category: string
      tags: string[]
      folder_id: number | null
      is_public: boolean
    }>,
  ) => http.put<MediaItem>(`/content/media/${id}`, data),
  remove: (id: number) => http.delete<null>(`/content/media/${id}`),
  batchDelete: (ids: number[]) =>
    http.post<{ affected: number }>('/content/media/batch/delete', {ids}),
  upload: (files: File[]) => {
    const form = new FormData()
    files.forEach((file) => form.append('files', file))
    return http.upload<{ files: MediaItem[] }>('/content/media/upload', form)
  },

  // ---- 文件夹 ----
  folders: () => http.page<MediaFolder>('/content/media/folders'),
  folderTree: () => http.get<MediaFolder[]>('/content/media/folders/tree'),
  createFolder: (data: { name: string; parent_id?: number | null; description?: string }) =>
    http.post<MediaFolder>('/content/media/folders', data),
  updateFolder: (id: number, data: { name?: string; parent_id?: number | null; description?: string }) =>
    http.put<MediaFolder>(`/content/media/folders/${id}`, data),
  removeFolder: (id: number) => http.delete<null>(`/content/media/folders/${id}`),
}
